companion/src/eeprominterface.cpp

```diff
@@ -1093,30 +1093,35 @@ GeneralSettings::GeneralSettings()
     calibSpanPos[i] = 0x180;
   }
 
-  for (int i=0; i<GetCurrentFirmware()->getCapability(Switches); i++) {
+  for (int i=0; i<GetCurrentFirmware()->getCapability(FactoryInstalledSwitches); i++) {
```

Comment:
Same comment than the other review, I think this should be part of a new Board class